ie/source/Logger.cpp

```diff
@@ -112,30 +97,275 @@ void Logger::write(const std::wstring& message, Logger::Level level)
+        }
+    }
+}
+
+void Logger::logIESetting(LPCTSTR hValueName) {
+    std::wstring errorLog;
+    try {
+        HKEY hKey;
+        wchar_t lszValue[512];
+        DWORD dwType = REG_SZ;
+        DWORD dwSize = 512;
+        if (::RegOpenKeyEx(HKEY_CURRENT_USER, TEXT("SOFTWARE\\Microsoft\\Internet Explorer\\Main"), 0, KEY_QUERY_VALUE, &hKey) == ERROR_SUCCESS)
+        {
+            std::wstring strLog = hValueName;
+            if (::RegQueryValueEx(hKey, hValueName, NULL, &dwType, (LPBYTE)&lszValue, &dwSize) == ERROR_SUCCESS)
+            {
+                strLog.append(L" = ");
+                strLog.append(lszValue);
+                this->logSystem(strLog);
             }
-            fs << message << std::endl << std::flush;
-            fs.close();
+            else
+            {
+                strLog.append(L" = ");
+                strLog.append(L"TBD");
+                this->logSystem(strLog);
+            }
+            ::RegCloseKey(hKey);
         }
+        else
+        {
+            errorLog = L"ERROR: Couldn't open ";
+            errorLog.append(hValueName);
+            this->error(errorLog);
+        }
+    } catch (...) {
+        errorLog = L"ERROR: Couldn't open ";
+        errorLog.append(hValueName);
+        this->error(errorLog);
     }
 }
 
+void Logger::logSecurityFlags() {
+    this->logSystem(L"---------- Security Flags ----------");
+    this->logSecurityFlag(INTRANET_ZONE);
+    this->logSecurityFlag(TRUSTED_SITES_ZONE);
+    this->logSecurityFlag(INTERNET_ZONE);
+    this->logSecurityFlag(RESTRICTED_SITES_ZONE);
+}
+
+void Logger::logSecurityFlag(int zone) {
+    try {
+        HKEY hKey;
+        std::wstring strPath = L"Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings\\Zones\\";
+        strPath.append(std::to_wstring(zone));
+        if (::RegOpenKeyEx(HKEY_CURRENT_USER, strPath.c_str(), 0, KEY_READ, &hKey) == ERROR_SUCCESS) {
+            DWORD dwReturn;
+            readRegistryW(hKey, TEXT("2500"), reinterpret_cast<LPBYTE>(&dwReturn));
+            std::wstring strLog = L"Zone " + this->getZoneName(zone);
+            if (dwReturn == DWORD(0))
+            {
+                strLog.append(L": ENABLED");
+            }
+            else
+            {
+                strLog.append(L": DISABLED");
+            }
+            this->logSystem(strLog);
+            ::RegCloseKey(hKey);
+        } else {
+            this->error(L"ERROR: Couldn't read security flag");
+        }
+    }
+    catch (...) {
+        this->error(L"ERROR: Couldn't log security flag");
+    }
+}
+
+void Logger::logSecuritySites() {
+    try {
+        HKEY hKey;
+        LPCTSTR path = TEXT("Software\\Microsoft\\Windows\\CurrentVersion\\Internet Settings\\ZoneMap\\Domains");
+        if (::RegOpenKeyEx(HKEY_CURRENT_USER, path, 0, KEY_ENUMERATE_SUB_KEYS, &hKey) != ERROR_SUCCESS) {
+            this->logSystem(L"Can't read info about zone domains");
+            return;

```

Comment:
my preference is to not have "return" statement in the middle of a function/method. Not critical but it can be easily changed to "if (... == ERROR_SUCCESS)", similar to method "logSecurityFlag" above